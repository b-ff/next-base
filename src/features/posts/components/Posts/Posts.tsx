"use client";
import { Flex } from "@/components/Flex";
import { ApiRoutes } from "@/enums/ApiRoutes";
import { Post as PostModel } from "@/schema";
import { FC, useEffect, useState } from "react";
import { Post } from "../Post";

export const Posts: FC = () => {
  const [isLoading, setIsLoading] = useState<boolean>(true);
  const [posts, setPosts] = useState<PostModel[]>([]);

  useEffect(() => {
    setIsLoading(true);
    fetch(ApiRoutes.Posts)
      .then((response) => response.json())
      .then(setPosts)
      .finally(() => setIsLoading(false));
  }, [setIsLoading, setPosts]);

  return isLoading ? (
    <>Loading...</>
  ) : posts.length ? (
    <Flex gap={2}>
      {posts.map((post: PostModel) => (
        <Post post={post} key={post.id} />
      ))}
    </Flex>
  ) : (
    <>There is no posts yet, come back later...</>
  );
};
