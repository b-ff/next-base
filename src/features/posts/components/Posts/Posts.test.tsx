import { Posts } from "./Posts";
import { render, waitFor } from "@testing-library/react";

beforeEach(() => {
  global.fetch = jest.fn(
    async () =>
      ({
        json: jest.fn(),
      } as any)
  );
});

describe("Component Posts", () => {
  it("Should render as expected", async () => {
    const { container } = render(<Posts />);
    await waitFor(() => expect(global.fetch).toBeCalled());
    expect(container).toMatchSnapshot();
  });
});
