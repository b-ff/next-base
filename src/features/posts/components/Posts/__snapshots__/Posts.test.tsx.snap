// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Posts Should render as expected 1`] = `
<div>
  Loading...
</div>
`;
